modules/fedimint-ln-common/src/api.rs

```diff
@@ -127,3 +141,52 @@ where
         }
     }
 }
+
+/// Filter out duplicate gateways. This is necessary because different guardians
+/// may have different TTLs for the same gateway, so two
+/// `LightningGatewayAnnouncement`s representing the same gateway registration
+/// may not be equal.
+fn filter_duplicate_gateways(
+    gateways: Vec<LightningGatewayAnnouncement>,
+) -> Vec<LightningGatewayAnnouncement> {
+    let gateways_by_gateway_id =
+        gateways
+            .into_iter()
+            .fold(std::collections::HashMap::new(), |mut acc, gateway| {
+                acc.entry(gateway.info.gateway_id)
+                    .or_insert_with(Vec::new)
+                    .push(gateway);
+                acc
+            });
```

Comment:
Could be replaced with [`Itertools::into_group_map`](https://docs.rs/itertools/latest/itertools/trait.Itertools.html#method.into_group_map).

Reply:
Very cool! Done.